tensorboard/plugins/profile/README.md

```diff
@@ -1,135 +1,89 @@
-    the TPU)
-*   preprocessing_us: string (time in us spent on data pre-processing)
-*   unclassified_nonenqueue_us: string (time in us spent on other components)
-
-2.2.2 Tabular data: rows are input-pipeline related tensorflow ops. column ids
-are defined below.
-
-*   opName: string
-*   count: number
-*   timeInMs: number
-*   timeInPercent: number
-*   selfTimeInMs: number
-*   selfTimeInPercent: number
-*   category: string
-
-#### 2.3 Recommendation DataTable
-
-2.3.1 Properties:
-
-*   overall: string (the overall recommendation to users)
-
-2.3.2 Tabular data: rows are recommendation details. column ids are defined
-below:
-
-*   link: string (recommendation for optimizing an individual input-processing
-    component)
+  The TensorBoard Profile Dashboard includes a suite of cloud TPU tools. These
+  tools help you understand, debug and optimize TensorFlow programs to run on
+  cloud TPUs.
+
+## Prerequisites
+  Before you can use the tools in Profile Dashboard, you must have access to
+  google cloud TPUs. You also need to capture trace information while your model
+  is running.
+
+## Overview Page
+  The Overview Page gives an overview of the performance of the workload
+  running on the TPU. It has five sections as shown in the following figure:
+
+  ![Overview Page](docs/overview-page.png)
+
+  * Performance Summary, which includes:
+
+    - The step time averaged over all steps sampled
+    - The percentage of Host time that is idle
+    - The percentage of TPU time that is idle
+    - The percentage utilization of the TPU matrix units
+
+  * Step-time Graph, which plots a graph of step time (in milliseconds) over
+  all the steps sampled.
+
+  * Top 10 TensorFlow operations on TPU, which shows the TensorFlow operations
+  executed on the TPU that consumes the majority of time.
+
+  * Run Environment, which includes:
+
+    - The number of hosts used
+    - The type of TPU used
+    - The number of TPU cores
+    - The training batch size
+    - Job information (both the build command and run command)
+    - Recommendation for Next Steps, which first reports if the workload is
+    input bounded. Then it suggests the next tools for reducing the bottleneck
+    (depending on whether it is the input time, the TPU time, or both).
+
+## Trace Viewer
+  Trace Viewer contains a timeline that shows various operations in your
+  TensorFlow model that the TPUs and host machine executed over time.
+
+  ![Trace Viewer](docs/trace-viewer.png)
+
+  The Timeline pane contains the following elements:
+
+  * A top bar, which contains various auxiliary controls.
+  * A time axis, which shows time relative to the beginning of the trace.
+  * Section and track labels. Each section contains multiple tracks and
+  has a triangle on the left that you can click to expand and collapse the
+  section. There is one section for every processing element in the system.
+  Sections and tracks will be explained in more detail below.
```

Comment:
I don't actually see any more explanation of sections and tracks in this doc right now, unless I'm missing something?

Reply:
Explanation added.